src/app/view/components/dot-theme-selector-dropdown/dot-theme-selector-dropdown.component.ts

```diff
@@ -0,0 +1,135 @@
+import { Component, forwardRef, Input, OnInit } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { DotTheme } from '@models/dot-edit-layout-designer';
+import { DotThemesService } from '@services/dot-themes/dot-themes.service';
+import { PaginatorService } from '@services/paginator';
+import { SiteService } from 'dotcms-js';
+import { LazyLoadEvent } from 'primeng/api';
+import { mergeMap, pluck, take } from 'rxjs/operators';
+
+@Component({
+    selector: 'dot-theme-selector-dropdown',
+    templateUrl: './dot-theme-selector-dropdown.component.html',
+    styleUrls: ['./dot-theme-selector-dropdown.component.scss'],
+    providers: [
+        {
+            multi: true,
+            provide: NG_VALUE_ACCESSOR,
+            useExisting: forwardRef(() => DotThemeSelectorDropdownComponent)
+        }
+    ]
+})
+export class DotThemeSelectorDropdownComponent implements OnInit, ControlValueAccessor {
+    themes: DotTheme[] = [];
+    value: DotTheme = null;
+    totalRecords: number = 0;
+    currentSiteIdentifier: string;
+    currentOffset: number;
+
+    @Input()
+    placeholder = 'Select Themes';
+
+    constructor(
+        private readonly paginatorService: PaginatorService,
+        private readonly siteService: SiteService,
+        private readonly themesService: DotThemesService
+    ) {}
+
+    ngOnInit(): void {
+        this.paginatorService.url = 'v1/themes';
+        this.paginatorService.paginationPerPage = 5;
+
+        this.siteService
+            .getCurrentSite()
+            .pipe(
+                take(1),
```

Comment:
I;m not sure if this `take(1)` is being applied to the `this.paginatorService.getWithOffset` as well.